astropy/table/table.py

```diff
@@ -870,9 +870,8 @@ def _init_from_list_of_dicts(self, data, names, dtype, n_cols, copy):
 
         # Fill the missing entries with first values
         if missing_indexes:
-            n_cols = len(names_from_data)
             first_vals = {}
-            for name in names_from_data:
+            for name in missing_indexes:
```

Comment:
This was iterating over every column and the `defaultdict(list)` was making an empty list, which then told the code later on to convert the column to MaskedColumn.  So every column was masked.  Oops.